server/service/handler.go

```diff
@@ -649,23 +649,18 @@ func attachFleetAPIRoutes(r *mux.Router, h *fleetHandlers) {
 }
 
 func attachNewStyleFleetAPIRoutes(r *mux.Router, svc fleet.Service, opts []kithttp.ServerOption) {
-	e := NewUserAuthenticatedEndpointer(svc, opts, r)
+	e := NewUserAuthenticatedEndpointer(svc, opts, r, "v1", "2021-11")
+
 	e.POST("/api/v1/fleet/users/roles/spec", applyUserRoleSpecsEndpoint, applyUserRoleSpecsRequest{})
 	e.POST("/api/v1/fleet/translate", translatorEndpoint, translatorRequest{})
 	e.POST("/api/v1/fleet/spec/teams", applyTeamSpecsEndpoint, applyTeamSpecsRequest{})
 	e.PATCH("/api/v1/fleet/teams/{team_id:[0-9]+}/secrets", modifyTeamEnrollSecretsEndpoint, modifyTeamEnrollSecretsRequest{})
 
 	// Alias /api/v1/fleet/team/ -> /api/v1/fleet/teams/
-	e.GET("/api/v1/fleet/team/{team_id}/schedule", getTeamScheduleEndpoint, getTeamScheduleRequest{})
-	e.POST("/api/v1/fleet/team/{team_id}/schedule", teamScheduleQueryEndpoint, teamScheduleQueryRequest{})
-	e.PATCH("/api/v1/fleet/team/{team_id}/schedule/{scheduled_query_id}", modifyTeamScheduleEndpoint, modifyTeamScheduleRequest{})
-	e.DELETE("/api/v1/fleet/team/{team_id}/schedule/{scheduled_query_id}", deleteTeamScheduleEndpoint, deleteTeamScheduleRequest{})
-	// End alias
-
-	e.GET("/api/v1/fleet/teams/{team_id}/schedule", getTeamScheduleEndpoint, getTeamScheduleRequest{})
-	e.POST("/api/v1/fleet/teams/{team_id}/schedule", teamScheduleQueryEndpoint, teamScheduleQueryRequest{})
-	e.PATCH("/api/v1/fleet/teams/{team_id}/schedule/{scheduled_query_id}", modifyTeamScheduleEndpoint, modifyTeamScheduleRequest{})
-	e.DELETE("/api/v1/fleet/teams/{team_id}/schedule/{scheduled_query_id}", deleteTeamScheduleEndpoint, deleteTeamScheduleRequest{})
+	e.GET("/api/v1/fleet/team[,s]/{team_id}/schedule", getTeamScheduleEndpoint, getTeamScheduleRequest{})
+	e.POST("/api/v1/fleet/team[,s]/{team_id}/schedule", teamScheduleQueryEndpoint, teamScheduleQueryRequest{})
```

Comment:
I also took a chance and simplified the aliases with a regex instead of the duplicate.

Reply:
Are we sure this is supported? I took a glance at the gorilla/mux docs and couldn't find anything about such templates (it's not really a regexp, as in a regexp that would only match `team,` with the comma and `teams` with the `s`) outside of the `{...}` variable placeholders, from the `Route.Path` docs:

> Path adds a matcher for the URL path. It accepts a template with zero or more URL variables enclosed by {}. The template must start with a "/". Variables can define an optional regexp pattern to be matched

Reply:
Yeah, I tried a few different combos: team, teams, teamss, works for the first two, fails on the second.